crescent/typedefs.py

```diff
@@ -1,13 +1,24 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Awaitable, Callable, Protocol, Sequence
+from typing import Any, Awaitable, Callable, Dict, Protocol, Sequence, Union
 
-if TYPE_CHECKING:
-    from .context import Context
+from hikari import PartialChannel, Role, User
 
-__all__: Sequence[str] = ("CommandCallback",)
+from crescent.commands.hooks import HookResult
+from crescent.context import Context
+from crescent.mentionable import Mentionable
+
+__all__: Sequence[str] = (
+    "CommandCallback",
+    "CommandOptionsT",
+    "ClassCommandProto",
+    "HookCallbackT",
+)
 
 CommandCallback = Callable[..., Awaitable[Any]]
+OptionTypesT = Union[str, bool, int, float, PartialChannel, Role, User, Mentionable]
+CommandOptionsT = Dict[str, OptionTypesT]
+HookCallbackT = Callable[[Context, CommandOptionsT], Awaitable[HookResult]]
```

Comment:
Looks like callbacks should be allowed to return None? That would make this `Awaitable[Optional[HookResult]]`.